esp32-firmware/src/main.cpp

```
@@ -26,39 +35,130 @@ const int MIN_ZERO_CROSSINGS = 2;           // Min 2 przejścia = 1 pełny okres
+        adc_digi_output_data_t *samples = (adc_digi_output_data_t *)dmaBuffer;
+        for (int i = 0; i < sampleCount; i++) {
+            uint8_t channel = samples[i].type1.channel;
+            uint16_t value = samples[i].type1.data;
+
+            if (channel == ADC_CHANNEL_U && idxU < SAMPLES) {
+                frameU[idxU++] = value;
+            } else if (channel == ADC_CHANNEL_I && idxI < SAMPLES) {
+                frameI[idxI++] = value;
+            }
+
+            if (idxU >= SAMPLES && idxI >= SAMPLES) {
```

Clang (14.0.6)

[note] +3, including nesting penalty of 2, nesting level increased to 3

(clang)